src/services/auth/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { Consent } from './service/identities/consents';
+
+export type ScopeTreeLeaf = {
+  scope: string;
+  atomically_revocable: boolean;
+  children: ScopeTreeLeaf[];
+};
+/**
+ * Parses a scope string into a normalized structure (leaf) for easier comparison.
+ */
+function parseScope(s: string): ScopeTreeLeaf {
+  let parsedScope = s;
+  const revocable = parsedScope.startsWith('*');
+  if (revocable) {
+    parsedScope = parsedScope.replace(/^\*\s*/, '');
+  }
+  let children: ScopeTreeLeaf[] = [];
+  /**
+   * If there is no bracket, then there are no children and we can return the parsed scope.
+   */
+  const firstBracket = parsedScope.indexOf('[');
+  if (firstBracket === -1) {
+    return {
+      scope: parsedScope,
+      atomically_revocable: revocable,
+      children: [],
+    };
+  }
+  /**
+   * The top-level scope is everything before the first encountered bracket.
+   */
+  const topLevelScope = parsedScope.slice(0, firstBracket);
+  /**
+   * The children are everything inside the brackets.
+   */
+  children = parsedScope
+    .slice(firstBracket + 1, -1)
+    .split(' ')
+    .map(parseScope);
+  return {
+    scope: topLevelScope,
+    atomically_revocable: revocable,
+    children,
+  };
+}
+
+/**
+ * Converts a scope string into a tree structure for easier comparison.
+ */
+export function toScopeTree(scope: string): ScopeTreeLeaf[] {
+  const scopes = scope.split(' ');
```

Comment:
Not sure it's totally necessary to address at this point, but worth pointing out that some scopes will possibly contain non-semantic spaces. The python sdk has some unit tests around this:
https://github.com/globus/globus-sdk-python/blob/d61420193674f97ec0d540633585b4635aba383b/tests/unit/scopes/test_scope_parser.py#L72

Reply:
This was a naive (incorrect) split – exact same issue we were discussing a few weeks ago trying to come up with a regex to solve for the deeply nested, space separated dependent scopes. 

I just pushed an update that adds a `splitScopeString` that is used internally here and no longer does the naive `.split(' ')` _or_ regex and instead just parses the string (I was seeing issues with what we came up with `/\s+(?=[^\]]*([[]|$))/`).

I've added some tests for `splitScopeString` that we could expand, but I think it covers the most common cases currently.